### PR TITLE
refactor(sqlsmith): refactor FROM generation

### PR DESCRIPTION
**Pull request by xxchan**
I hereby agree to the terms of the [RisingWave Labs, Inc. Contributor License Agreement](https://gist.github.com/TennyZhuang/f00be7f16996ea48effb049aa7be4d66#file-rw_cla).

## What's changed and what's your intention?
After the refactoring, any `TableFactor` can occur in a join.

## Checklist

- [ ] I have written necessary rustdoc comments
- [ ] I have added necessary unit tests and integration tests
- [ ] I have added fuzzing tests or opened an issue to track them. (Optional, recommended for new SQL features #7934).
- [ ] My PR contains breaking changes. (If it deprecates some features, please create a tracking issue to remove them in the future).
- [ ] All checks passed in `./risedev check` (or alias, `./risedev c`)
- [ ] My PR changes performance-critical code. (Please run macro/micro-benchmarks and show the results.)

## Documentation

- [ ] My PR contains user-facing changes.

<details><summary>Click here for Documentation</summary>

### Types of user-facing changes

Please keep the types that apply to your changes, and remove the others.

- Installation and deployment
- Connector (sources & sinks)
- SQL commands, functions, and operators
- RisingWave cluster configuration changes
- Other (please specify in the release note below)

### Release note

</details>

### CODE REVIEW

**Comment by xxchan** [comment]
> not sure why such a query is generated

```sql
CREATE MATERIALIZED VIEW m0 AS 
SELECT tumble_0.c3 AS col_0, ((964)) AS col_1, ((BIGINT '738') + (BIGINT '6561859126141167501')) AS col_2 
FROM 
	tumble(alltypes1, alltypes1.c11, INTERVAL '6') AS tumble_0 
JOIN 
	hop(alltypes1, alltypes1.c11, INTERVAL '153575', INTERVAL '5989425') AS hop_1 
ON c5 = c5 AND true 
WHERE tumble_0.c1 
GROUP BY tumble_0.c15, hop_1.c16, tumble_0.c1, hop_1.c8, tumble_0.c6, tumble_0.c8, tumble_0.c3, hop_1.c7, tumble_0.c2, hop_1.c1;
```

I think it looks correct after the change, except table names are not specified in the `ON` clause 🤔.

**Comment by kwannoel** [comment]
LGTM after #10348 merged into this branch.

**Review by kwannoel — APPROVED** [review]

**Pull request merged**
